From a630a10d74f1f28dc8668876dadf526db36b9fe3 Mon Sep 17 00:00:00 2001
From: Philippe De Swert <philippedeswert@gmail.com>
Date: Fri, 12 Sep 2014 16:49:48 +0300
Subject: [PATCH] core: smack-setup: Actually allow for succesfully loading
 CIPSO policy

The line under the last switch statement  *loaded_policy = true;
would never be executed. As all switch cases return 0. Thus the
policy would never be marked as loaded.

Found with Coverity. Fixes: CID#1237785

(cherry picked from commit b9289d4c6e13ec5fb67bfce69c826d93b004da6a)
---
 src/core/smack-setup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/smack-setup.c b/src/core/smack-setup.c
index 8838d31a88..c2acba75c2 100644
--- a/src/core/smack-setup.c
+++ b/src/core/smack-setup.c
@@ -148,7 +148,7 @@ int smack_setup(void) {
                 return 0;
         case 0:
                 log_info("Successfully loaded Smack/CIPSO policies.");
-                return 0;
+                break;
         default:
                 log_warning("Failed to load Smack/CIPSO access rules: %s, ignoring.",
                             strerror(abs(r)));
